src/lib/ares_android.c

```diff
@@ -80,6 +81,30 @@ static jmethodID jni_get_method_id(JNIEnv *env, jclass cls,
   return mid;
 }
 
+static char *get_thread_name(char *name)
+{
+  if (prctl(PR_GET_NAME, name) == 0) {
+    return name;
+  }
+  return NULL;
+}
+
+static JavaVMAttachArgs get_jvm_attach_args(char *name)
+{
+  JavaVMAttachArgs args;
+  args.version = JNI_VERSION_1_6;
+  args.name    = get_thread_name(name);
+  args.group   = NULL;
+  return args;
+}
+
+static int jvm_attach(JNIEnv **env)
+{
+  char             name[17] = {0};
+  JavaVMAttachArgs args     = get_jvm_attach_args(name);
+  return (*android_jvm)->AttachCurrentThread(android_jvm, env, &args);
+}
+
```

Comment:
honestly, I'd just combine jvm_attach(), get_jvm_attach_args(), and get_thread_name() into a single function.  They're only used together and it doesn't really enhance readability.  But otherwise, yeah, this is good.

Reply:
Combined